Plugin~/Src/MeshSync/SceneGraph/msEntityConverter.cpp

```diff
@@ -140,6 +142,15 @@ void ScaleConverter::convertAnimationCurve(AnimationCurve &c)
     }
 }
 
+void ScaleConverter::convertInstanceInfos(InstanceInfo& v)
+{
+    for (size_t i = 0; i < v.transforms.size(); ++i)
+    {
+        // We can divide the w component instead of applying the multiplier on xyz:
+        v.transforms[i][3][3] /= m_scale;
+    }
```

Comment:
might be good to check if m_scale is 1 and avoid the loop if so. Many users might not change this

Reply:
Where is this invoked? I don't see any callback

Reply:
may also need to check if the scale is 0 ?

Reply:
I've added a check for 0. Maybe we should not allow 0 in general, though?

Reply:
> might be good to check if m_scale is 1 and avoid the loop if so. Many users might not change this

There's already a check for 1 in the code that calls this.

Reply:
I think he meant that we should probably put the check in this function ?
Or otherwise we will have to duplicate that code if we want to call this function from a new place.

Reply:
None of the other converters do that. It's an optimisation to not run the converters on all entities because the output would not change. Do we really want to slow this down by adding unnecessary code for each entity to check for the same thing again?

Reply:
Hm, I haven't checked the caller, but now I see it.

Ok, since this PR has the check against 0, I thought that this should have the check against 1 as well.
If we want to optimize this, then we should check the scale against both 0 and 1 on the caller side, but not here, like you said.
In other words, please remove the 0 check here.

Another thought is that, since this code is currently only called in MeshSyncDCCPlugins, could you please duplicate the code there for now, so that we can release this fix quicker ?
Otherwise, we would have to increase the major/minor version of MeshSync, and that takes quite some time of preparation, and more testing as well.

Reply:
Ok, done here: https://github.com/Unity-Technologies/MeshSyncDCCPlugins/pull/237/files